### ISSUE TITLE
Release 4.1.0

### ISSUE THREAD

**Issue by ronyeh** [0xfe/vexflow#1407]
Now that we have Leland in VexFlow, we can make a 4.1 release.

@rvilarl @AaronDavidNewman can you help test out the current head revision?

The release script has been going pretty smoothly. After this release, I'd like to hand it over to @0xfe so he can do the next couple of releases, just to make sure there aren't any bugs with our release procedure.

**Comment by rvilarl** [comment]
https://github.com/stringsync/vexml

**Comment by ronyeh** [comment]
Thanks Rodrigo. I looked at it and also the original comment (in March) that spawned the idea for this new repo. I think musicxml support for VexFlow would be great. Good luck with that project! 

I'll go ahead and look at the noteheads PR, since it'll support the progress of your vexml project. It's a nice project name that is clear for any VexFlow user. Though I think the VeXML README and demos need to be fleshed out a bit. ;-)

**Comment by AaronDavidNewman** [comment]
I downloaded the latest master branch and used the build output directly from Smoosic, and I was able to use the Leland font.  Everything looked good, I also ran the unit tests and everything looked good.

**Comment by rvilarl** [comment]
It also works fine with OSMD.🙂

**Comment by ronyeh** [comment]
I verified that the release script works (with the patch removed). I just published a pre-release version to NPM:

**4.1.0-alpha.0**

The command I used was:
```
grunt release:alpha
```

It required me to input my 2FA token for NPM.

**Comment by rvilarl** [comment]
Would it make sense to have a 'alpha' branch? It is taking very long to get PRs merged and with that we could speed up on testing the changes with OSMD, ...

**Comment by mscuthbert** [comment]
I might, but I haven't done it yet, so not comfortable doing so unless ronyeh and 0xfe don't reply to this ping.

**Comment by ronyeh** [comment]
I can do a beta release today. If we don't find any bugs this week I can
push a full 4.1.0 out later this week.

**Comment by ronyeh** [comment]
I just shipped a beta:

https://www.npmjs.com/package/vexflow/v/4.1.0-beta.0

```
$ npm install vexflow@beta


changed 1 package, and audited 90 packages in 1s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```


@mscuthbert @rvilarl @AaronDavidNewman please test this beta build with your websites / apps. If it looks good, let me know and I can ship 4.1.0 later this week.

Thanks!

**Comment by rvilarl** [comment]
@ronyeh looks good with vexml

**Comment by ronyeh** [comment]
Thanks for testing with vexml. I'll integrate with my sample apps and go through my tests. Will hopefully release it before Monday.

**Comment by ronyeh** [comment]
I'm going to push a second beta because there has been a recent PR merge. After this new beta, let's hold off on new merges until 4.1 goes live. :-)

**Comment by mscuthbert** [comment]
It might be good when betas start coming out to base off of a v4.1 branch, etc. and release from specific branches rather than master.  I tend to do that on my sites.  Also lets us work hard on an incompatible next version but still patch major bugs on the most recent release.

**Comment by ronyeh** [comment]
Yeah, I agree that is the right way to go about releases. However we've
been keeping things lightweight so far by releasing from master. If someone
wants to investigate the release-it script to support easy releases from a
branch, please help us out 😁.

For now, we don't have so many PRs getting merged daily that I think it's
simpler (for me) to ship from master. If it becomes a problem, we can
switch over to the better way.

**Comment by mscuthbert** [comment]
Sounds good.  It can also just be a tag: "ready to merge" that we use during beta processes so we know as soon as we release we can hit merge on anything else.

**Comment by rvilarl** [comment]
The beta.1 is still not there. I have merge #1478.
Release 4.1.0 is now six months old, I suggest that we release and fix with a 4.1.1 if there is a problem.

**Comment by ronyeh** [comment]
Hi! Apologies for the delay. I spent some time updating the demos folder to work with the newly included Leland font. I was also testing to make sure none of the existing demos were broken.

I'll go ahead and release a RC now. I'm not going to push a full release tonight because there was a fresh PR merge. I'm pretty conservative with releases, especially if we have a fresh merge.

**Comment by ronyeh** [comment]
For example, with the most recent merge, running `grunt` produces a bunch of new docs/api files. Does that mean the API changed? I guess it makes sense because it was a refactoring PR. Hopefully it was only internal APIs?

Here were the docs that were generated when I just ran `grunt` in preparation for a new 4.1.0 release candidate.

```
	modified:   docs/api/assets/search.js
	modified:   docs/api/classes/BarNote.html
	modified:   docs/api/classes/ClefNote.html
	modified:   docs/api/classes/Crescendo.html
	modified:   docs/api/classes/GhostNote.html
	modified:   docs/api/classes/GlyphNote.html
	modified:   docs/api/classes/GraceNote.html
	modified:   docs/api/classes/GraceTabNote.html
	modified:   docs/api/classes/KeySigNote.html
	modified:   docs/api/classes/Note.html
	modified:   docs/api/classes/NoteHead.html
	modified:   docs/api/classes/RepeatNote.html
	modified:   docs/api/classes/StaveNote.html
	modified:   docs/api/classes/StemmableNote.html
	modified:   docs/api/classes/TabNote.html
	modified:   docs/api/classes/TextDynamics.html
	modified:   docs/api/classes/TextNote.html
	modified:   docs/api/classes/TimeSigNote.html
	modified:   docs/api/interfaces/GlyphProps.html
	modified:   docs/api/interfaces/KeyProps.html
```

Anyways, I'm not comfortable pushing a release to npm given these recent changes. I'll post here when the RC is live, and if you and Aaron sign off on it, I can ship the release (assuming no new PRs are merged).

**Comment by rvilarl** [comment]
Hi Ron! Thanks! 

I realised yesterday also that tools/fonts/inspectors/glyphs.html has two bugs:
- Leland is missing
- `const glyphs = font.getGlyphs();` should be `const glyphs = font[0].getGlyphs();`

**Comment by ronyeh** [comment]
You can submit a PR for this, but we could not merge it until 4.1.1, since it seems like a low priority issue.

**Comment by ronyeh** [comment]
Hi all, I put up a release candidate.

```
npm install vexflow@rc
```

Please test, and comment below if everything looks good on your end. I can ship 4.1.0 if I get at least two OKs from the folks on this thread. :-)

**Comment by ronyeh** [comment]
> * should be `const glyphs = font[0].getGlyphs();`
@rvilarl I actually think it should still be font.getGlyphs() since font is singular. Instead, in a previous line we should add the [0] so it looks like:

```
const font = Vex.Flow.setMusicFont(fontName)[0];
```

Yeah it's a bit ugly, but it is more consistent with the commented-out 3.0.9 code which extracts a single font object.

Anyways, open a PR for this and we can merge it after 4.1.0 is released.

**Comment by ronyeh** [comment]
We have shipped VexFlow 4.1.0!

Thanks Rodrigo @rvilarl for taking the lead on this release, and Aaron @AaronDavidNewman for contributing PRs and code reviews.

I documented the steps I took here: https://github.com/0xfe/vexflow/wiki/ReleaseSteps#release-vexflow-410
